File: archive/0_Other/Python/selectionSort.py
def selectionSort(arr):
    for i in range(len(arr)):
        s = i
        for j in range(i+1, len(arr)):
            if arr[s] > arr[j]:
                s = j
        arr[i], arr[s] = arr[s], arr[i]
    return arr


A = [64, 25, 12, 22, 11]
print(selectionSort(A))

File: archive/go/src/sort/selectionSort.go
package sort

func selectionSort(arr []int) {
	l := len(arr)
	if l < 2 {
		return
	}

	for i := 0; i < l-1; i++ {
		minIndex := i
		for j := i + 1; j < l; j++ {
			if arr[j] < arr[minIndex] {
				minIndex = j
			}
		}
		if i != minIndex {
			arr[i], arr[minIndex] = arr[minIndex], arr[i]
		}
	}
}
